prombench/Makefile

```diff
@@ -37,50 +30,96 @@ cluster_delete:
 		-v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
 		-f manifests/cluster_${PROVIDER}.yaml
 
+# /prombench <...> --bench.directory
+BENCHMARK_DIRECTORY := $(if $(BENCHMARK_DIRECTORY),$(BENCHMARK_DIRECTORY),manifests/prombench)
+# /prombench <...> --bench.version
+BENCHMARK_VERSION := $(if $(BENCHMARK_VERSION),$(BENCHMARK_VERSION),master)
+PROMBENCH_GIT_REPOSITORY ?= git@github.com:prometheus/test-infra.git
+PROMBENCH_DIR ?= .
+
+# maybe_pull_custom_version allows custom benchmarking as designed in
+# https://github.com/prometheus/proposals/pull/41. It allows calling
+# /prombench <release> --bench.version=<@commit or branch> which will cause
+# prombench GH job on Prometheus repo to call infra CLI with the non-master BENCHMARK_VERSION.
+# In such a case we pull a prombench repository for the given branch or commit version
+# and adjust PROMBENCH_DIR. As a result `make deploy` and `make clean` jobs
+# will apply /manifests/ apply custom manifests or even node pools.
+.PHONE: maybe_pull_custom_version
+maybe_pull_custom_version:
+ifeq (${BENCHMARK_VERSION},master)
+	@echo ">> Using standard benchmark configuration, from the docker image"
+else
+	@echo ">> Git pulling custom benchmark configuration from the ${BENCHMARK_VERSION}"
+	@$(eval $@_TMP_DIR=$(shell mktemp -d -t "prombench"))
+	cd ${$@_TMP_DIR} && git clone ${PROMBENCH_GIT_REPOSITORY}
+ifeq ($(subst @,,${BENCHMARK_VERSION}),${BENCHMARK_VERSION})
```

Comment:
We rely on the presence of the `@` character to tell if we're receiving a branch or a commit SHA? I do not understand this part 🤔

Reply:
Ok I found my own answer: https://github.com/prometheus/proposals/pull/41/files#diff-b472a523ad2ffdc696851b117f2ee2808d94166c886bf6f090294dc2e728e5e7R106

Indeed you mention the usage of `@` character

Reply:
Correct. This follows docker image reference format. Allows us to never confuse branch and commit (I doubt you can create branches with `@` chars - and even if you can we can block that).